orchestrator/src/compression/stateless/utils.rs

```
@@ -0,0 +1,90 @@
+use crate::compression::stateless::constants::MAX_N_BITS;
+use color_eyre::eyre::{bail, eyre};
+use itertools::Itertools;
+use num_bigint::BigUint;
+use num_traits::Zero;
+use starknet_core::types::Felt;
+
+/// Revert signatures (no Result) and use expect/panic
+pub fn get_n_elms_per_felt(elm_bound: u32) -> color_eyre::Result<usize> {
+    if elm_bound == 0 {
+        return Err(eyre!("Element bound cannot be 0"));
+    }
+    if elm_bound <= 1 {
+        return Ok(MAX_N_BITS);
+    }
+    let n_bits_required = (elm_bound - 1).ilog2() + 1;
+    Ok(MAX_N_BITS
+        / usize::try_from(n_bits_required)
+            .map_err(|err| eyre!("Failed usize conversion for bits required: {}", err))?)
+}
+
+/// Packs a slice of usize into a vector of Felt
+pub fn pack_usize_in_felts(elms: &[usize], elm_bound: u32) -> color_eyre::Result<Vec<Felt>> {
+    if elm_bound == 0 {
+        return Err(eyre!("Element bound cannot be 0 for packing"));
+    }
+    // Check elements are within bound
+    for elm in elms {
+        let elm_u32 = u32::try_from(*elm).map_err(|err| eyre!("Cannot convert element to u32: {}", err))?;
+        assert!(elm_u32 < elm_bound, "Element {} exceeds bound {}", elm, elm_bound);
```

let's throw a proper error instead!

---

Done